<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1255</width>
    <height>600</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
    <horstretch>5</horstretch>
    <verstretch>5</verstretch>
   </sizepolicy>
  </property>
  <property name="minimumSize">
   <size>
    <width>0</width>
    <height>600</height>
   </size>
  </property>
  <property name="maximumSize">
   <size>
    <width>1255</width>
    <height>600</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>Force-directed graph drawing</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QGridLayout" name="gridLayout">
    <item row="0" column="1">
     <widget class="QLabel" name="labelStart">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="font">
       <font>
        <pointsize>10</pointsize>
        <weight>75</weight>
        <bold>true</bold>
       </font>
      </property>
      <property name="text">
       <string>Start</string>
      </property>
     </widget>
    </item>
    <item row="0" column="2">
     <widget class="QLabel" name="labelResult">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="minimumSize">
       <size>
        <width>0</width>
        <height>0</height>
       </size>
      </property>
      <property name="maximumSize">
       <size>
        <width>16777215</width>
        <height>100</height>
       </size>
      </property>
      <property name="font">
       <font>
        <pointsize>10</pointsize>
        <weight>75</weight>
        <bold>true</bold>
       </font>
      </property>
      <property name="text">
       <string>Result: Step 0</string>
      </property>
     </widget>
    </item>
    <item row="1" column="0">
     <widget class="QGroupBox" name="groupBox_InitialG">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="minimumSize">
       <size>
        <width>150</width>
        <height>250</height>
       </size>
      </property>
      <property name="styleSheet">
       <string notr="true">QGroupBox{
border: 2px solid gray;
border-radius: 10px;
padding: 0.8px;
background: white;
selection-background-color: darkgray;
}

QGroupBox::title {
     subcontrol-origin: margin;
     subcontrol-position: top center; /* position at the top center */
     padding: 0 3px;
	margin-top: 0px;
 }</string>
      </property>
      <property name="title">
       <string>Initial binary tree</string>
      </property>
      <widget class="QTextEdit" name="textEdit_h">
       <property name="geometry">
        <rect>
         <x>60</x>
         <y>20</y>
         <width>61</width>
         <height>21</height>
        </rect>
       </property>
       <property name="sizePolicy">
        <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="inputMethodHints">
        <set>Qt::ImhDigitsOnly</set>
       </property>
       <property name="verticalScrollBarPolicy">
        <enum>Qt::ScrollBarAlwaysOff</enum>
       </property>
       <property name="horizontalScrollBarPolicy">
        <enum>Qt::ScrollBarAlwaysOff</enum>
       </property>
       <property name="html">
        <string>&lt;!DOCTYPE HTML PUBLIC &quot;-//W3C//DTD HTML 4.0//EN&quot; &quot;http://www.w3.org/TR/REC-html40/strict.dtd&quot;&gt;
&lt;html&gt;&lt;head&gt;&lt;meta name=&quot;qrichtext&quot; content=&quot;1&quot; /&gt;&lt;style type=&quot;text/css&quot;&gt;
p, li { white-space: pre-wrap; }
&lt;/style&gt;&lt;/head&gt;&lt;body style=&quot; font-family:'Sans Serif'; font-size:9pt; font-weight:400; font-style:normal;&quot;&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;5&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
       </property>
      </widget>
      <widget class="QLabel" name="label">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>20</y>
         <width>41</width>
         <height>21</height>
        </rect>
       </property>
       <property name="text">
        <string>Height</string>
       </property>
      </widget>
      <widget class="QLabel" name="label_2">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>40</y>
         <width>41</width>
         <height>21</height>
        </rect>
       </property>
       <property name="text">
        <string>|V| = </string>
       </property>
      </widget>
      <widget class="QTextEdit" name="textEdit_nV">
       <property name="enabled">
        <bool>false</bool>
       </property>
       <property name="geometry">
        <rect>
         <x>60</x>
         <y>40</y>
         <width>61</width>
         <height>21</height>
        </rect>
       </property>
       <property name="inputMethodHints">
        <set>Qt::ImhDigitsOnly</set>
       </property>
       <property name="verticalScrollBarPolicy">
        <enum>Qt::ScrollBarAlwaysOff</enum>
       </property>
       <property name="horizontalScrollBarPolicy">
        <enum>Qt::ScrollBarAlwaysOff</enum>
       </property>
       <property name="html">
        <string>&lt;!DOCTYPE HTML PUBLIC &quot;-//W3C//DTD HTML 4.0//EN&quot; &quot;http://www.w3.org/TR/REC-html40/strict.dtd&quot;&gt;
&lt;html&gt;&lt;head&gt;&lt;meta name=&quot;qrichtext&quot; content=&quot;1&quot; /&gt;&lt;style type=&quot;text/css&quot;&gt;
p, li { white-space: pre-wrap; }
&lt;/style&gt;&lt;/head&gt;&lt;body style=&quot; font-family:'Sans Serif'; font-size:9pt; font-weight:400; font-style:normal;&quot;&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;31&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
       </property>
      </widget>
      <widget class="QPushButton" name="pButton_generateG">
       <property name="geometry">
        <rect>
         <x>30</x>
         <y>70</y>
         <width>81</width>
         <height>23</height>
        </rect>
       </property>
       <property name="text">
        <string>generate</string>
       </property>
      </widget>
      <widget class="QPushButton" name="pbuttonSave_InitialG">
       <property name="geometry">
        <rect>
         <x>30</x>
         <y>100</y>
         <width>81</width>
         <height>23</height>
        </rect>
       </property>
       <property name="text">
        <string>Save img</string>
       </property>
      </widget>
      <widget class="QGroupBox" name="groupBox_Algorithm">
       <property name="enabled">
        <bool>false</bool>
       </property>
       <property name="geometry">
        <rect>
         <x>0</x>
         <y>150</y>
         <width>150</width>
         <height>100</height>
        </rect>
       </property>
       <property name="sizePolicy">
        <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="minimumSize">
        <size>
         <width>150</width>
         <height>100</height>
        </size>
       </property>
       <property name="maximumSize">
        <size>
         <width>150</width>
         <height>100</height>
        </size>
       </property>
       <property name="styleSheet">
        <string notr="true">QGroupBox{
border: 2px solid gray;
border-radius: 10px;
padding: 0.8px;
background: white;
selection-background-color: darkgray;
}

QGroupBox::title {
     subcontrol-origin: margin;
     subcontrol-position: top center; /* position at the top center */
     padding: 0 3px;
	margin-top: 0px;
 }</string>
       </property>
       <property name="title">
        <string>Algorithm</string>
       </property>
       <widget class="QRadioButton" name="rB_KamadaKawai">
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>30</y>
          <width>121</width>
          <height>21</height>
         </rect>
        </property>
        <property name="text">
         <string>Kamada_Kawai</string>
        </property>
        <property name="checked">
         <bool>true</bool>
        </property>
       </widget>
       <widget class="QRadioButton" name="rB_HarelKoren">
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>50</y>
          <width>121</width>
          <height>21</height>
         </rect>
        </property>
        <property name="text">
         <string>Harel_Koren</string>
        </property>
       </widget>
      </widget>
      <widget class="QCheckBox" name="cBox_ShowLabels">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>125</y>
         <width>111</width>
         <height>21</height>
        </rect>
       </property>
       <property name="text">
        <string>show labels</string>
       </property>
       <property name="checked">
        <bool>true</bool>
       </property>
      </widget>
     </widget>
    </item>
    <item row="1" column="1" rowspan="2">
     <widget class="matplotlibWidget" name="MatplotlibWidget1" native="true">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="minimumSize">
       <size>
        <width>530</width>
        <height>530</height>
       </size>
      </property>
      <property name="styleSheet">
       <string notr="true">background-color: rgb(255, 255, 255);</string>
      </property>
     </widget>
    </item>
    <item row="1" column="2" rowspan="2">
     <widget class="matplotlibWidget" name="MatplotlibWidget2" native="true">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="minimumSize">
       <size>
        <width>530</width>
        <height>530</height>
       </size>
      </property>
      <property name="styleSheet">
       <string notr="true">background-color: rgb(255, 255, 255);</string>
      </property>
     </widget>
    </item>
    <item row="2" column="0">
     <widget class="QGroupBox" name="groupBox_GDrawing">
      <property name="enabled">
       <bool>true</bool>
      </property>
      <property name="sizePolicy">
       <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="minimumSize">
       <size>
        <width>150</width>
        <height>230</height>
       </size>
      </property>
      <property name="maximumSize">
       <size>
        <width>150</width>
        <height>270</height>
       </size>
      </property>
      <property name="styleSheet">
       <string notr="true">QGroupBox{
border: 2px solid gray;
border-radius: 10px;
padding: 0.8px;
background: white;
selection-background-color: darkgray;
}

QGroupBox::title {
     subcontrol-origin: margin;
     subcontrol-position: top center; /* position at the top center */
     padding: 0 3px;
	margin-top: 0px;
 }</string>
      </property>
      <property name="title">
       <string>Draw graph</string>
      </property>
      <widget class="QPushButton" name="pbuttonStart">
       <property name="enabled">
        <bool>false</bool>
       </property>
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>20</y>
         <width>61</width>
         <height>23</height>
        </rect>
       </property>
       <property name="text">
        <string>Run</string>
       </property>
      </widget>
      <widget class="QPushButton" name="pbuttonStep">
       <property name="enabled">
        <bool>false</bool>
       </property>
       <property name="geometry">
        <rect>
         <x>80</x>
         <y>20</y>
         <width>61</width>
         <height>23</height>
        </rect>
       </property>
       <property name="text">
        <string>Step</string>
       </property>
      </widget>
      <widget class="QPushButton" name="pbuttonSave">
       <property name="enabled">
        <bool>false</bool>
       </property>
       <property name="geometry">
        <rect>
         <x>40</x>
         <y>80</y>
         <width>61</width>
         <height>23</height>
        </rect>
       </property>
       <property name="text">
        <string>Save img</string>
       </property>
      </widget>
      <widget class="QPushButton" name="pbuttonContinue">
       <property name="enabled">
        <bool>false</bool>
       </property>
       <property name="geometry">
        <rect>
         <x>80</x>
         <y>50</y>
         <width>61</width>
         <height>23</height>
        </rect>
       </property>
       <property name="text">
        <string>Continue</string>
       </property>
      </widget>
      <widget class="QGroupBox" name="groupBox_Param">
       <property name="geometry">
        <rect>
         <x>0</x>
         <y>120</y>
         <width>150</width>
         <height>150</height>
        </rect>
       </property>
       <property name="sizePolicy">
        <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="minimumSize">
        <size>
         <width>150</width>
         <height>0</height>
        </size>
       </property>
       <property name="maximumSize">
        <size>
         <width>150</width>
         <height>150</height>
        </size>
       </property>
       <property name="styleSheet">
        <string notr="true">QGroupBox{
border: 2px solid gray;
border-radius: 10px;
padding: 0.8px;
background: white;
selection-background-color: darkgray;
}

QGroupBox::title {
     subcontrol-origin: margin;
     subcontrol-position: top center; /* position at the top center */
     padding: 0 3px;
	margin-top: 0px;
 }</string>
       </property>
       <property name="title">
        <string>Parameters</string>
       </property>
       <widget class="QLabel" name="label_L0">
        <property name="geometry">
         <rect>
          <x>20</x>
          <y>40</y>
          <width>41</width>
          <height>21</height>
         </rect>
        </property>
        <property name="text">
         <string>L_0 = </string>
        </property>
       </widget>
       <widget class="QTextEdit" name="textEdit_K">
        <property name="geometry">
         <rect>
          <x>70</x>
          <y>20</y>
          <width>61</width>
          <height>21</height>
         </rect>
        </property>
        <property name="sizePolicy">
         <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="inputMethodHints">
         <set>Qt::ImhDigitsOnly</set>
        </property>
        <property name="verticalScrollBarPolicy">
         <enum>Qt::ScrollBarAlwaysOff</enum>
        </property>
        <property name="horizontalScrollBarPolicy">
         <enum>Qt::ScrollBarAlwaysOff</enum>
        </property>
        <property name="html">
         <string>&lt;!DOCTYPE HTML PUBLIC &quot;-//W3C//DTD HTML 4.0//EN&quot; &quot;http://www.w3.org/TR/REC-html40/strict.dtd&quot;&gt;
&lt;html&gt;&lt;head&gt;&lt;meta name=&quot;qrichtext&quot; content=&quot;1&quot; /&gt;&lt;style type=&quot;text/css&quot;&gt;
p, li { white-space: pre-wrap; }
&lt;/style&gt;&lt;/head&gt;&lt;body style=&quot; font-family:'Sans Serif'; font-size:9pt; font-weight:400; font-style:normal;&quot;&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;1&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
        </property>
       </widget>
       <widget class="QLabel" name="label_K">
        <property name="geometry">
         <rect>
          <x>20</x>
          <y>20</y>
          <width>41</width>
          <height>21</height>
         </rect>
        </property>
        <property name="text">
         <string>K = </string>
        </property>
       </widget>
       <widget class="QTextEdit" name="textEdit_L0">
        <property name="geometry">
         <rect>
          <x>70</x>
          <y>40</y>
          <width>61</width>
          <height>21</height>
         </rect>
        </property>
        <property name="inputMethodHints">
         <set>Qt::ImhDigitsOnly</set>
        </property>
        <property name="verticalScrollBarPolicy">
         <enum>Qt::ScrollBarAlwaysOff</enum>
        </property>
        <property name="horizontalScrollBarPolicy">
         <enum>Qt::ScrollBarAlwaysOff</enum>
        </property>
        <property name="html">
         <string>&lt;!DOCTYPE HTML PUBLIC &quot;-//W3C//DTD HTML 4.0//EN&quot; &quot;http://www.w3.org/TR/REC-html40/strict.dtd&quot;&gt;
&lt;html&gt;&lt;head&gt;&lt;meta name=&quot;qrichtext&quot; content=&quot;1&quot; /&gt;&lt;style type=&quot;text/css&quot;&gt;
p, li { white-space: pre-wrap; }
&lt;/style&gt;&lt;/head&gt;&lt;body style=&quot; font-family:'Sans Serif'; font-size:9pt; font-weight:400; font-style:normal;&quot;&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;1&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
        </property>
       </widget>
       <widget class="QLabel" name="label_eps">
        <property name="geometry">
         <rect>
          <x>20</x>
          <y>60</y>
          <width>41</width>
          <height>21</height>
         </rect>
        </property>
        <property name="text">
         <string>eps</string>
        </property>
       </widget>
       <widget class="QTextEdit" name="textEdit_eps">
        <property name="geometry">
         <rect>
          <x>70</x>
          <y>60</y>
          <width>61</width>
          <height>21</height>
         </rect>
        </property>
        <property name="inputMethodHints">
         <set>Qt::ImhDigitsOnly</set>
        </property>
        <property name="verticalScrollBarPolicy">
         <enum>Qt::ScrollBarAlwaysOff</enum>
        </property>
        <property name="horizontalScrollBarPolicy">
         <enum>Qt::ScrollBarAlwaysOff</enum>
        </property>
        <property name="html">
         <string>&lt;!DOCTYPE HTML PUBLIC &quot;-//W3C//DTD HTML 4.0//EN&quot; &quot;http://www.w3.org/TR/REC-html40/strict.dtd&quot;&gt;
&lt;html&gt;&lt;head&gt;&lt;meta name=&quot;qrichtext&quot; content=&quot;1&quot; /&gt;&lt;style type=&quot;text/css&quot;&gt;
p, li { white-space: pre-wrap; }
&lt;/style&gt;&lt;/head&gt;&lt;body style=&quot; font-family:'Sans Serif'; font-size:9pt; font-weight:400; font-style:normal;&quot;&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;0.001&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
        </property>
       </widget>
       <widget class="QPushButton" name="pbuttonReset">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="geometry">
         <rect>
          <x>40</x>
          <y>120</y>
          <width>61</width>
          <height>23</height>
         </rect>
        </property>
        <property name="text">
         <string>Reset</string>
        </property>
       </widget>
       <widget class="QTextEdit" name="textEdit_maxit">
        <property name="geometry">
         <rect>
          <x>70</x>
          <y>80</y>
          <width>61</width>
          <height>21</height>
         </rect>
        </property>
        <property name="inputMethodHints">
         <set>Qt::ImhDigitsOnly</set>
        </property>
        <property name="verticalScrollBarPolicy">
         <enum>Qt::ScrollBarAlwaysOff</enum>
        </property>
        <property name="horizontalScrollBarPolicy">
         <enum>Qt::ScrollBarAlwaysOff</enum>
        </property>
        <property name="html">
         <string>&lt;!DOCTYPE HTML PUBLIC &quot;-//W3C//DTD HTML 4.0//EN&quot; &quot;http://www.w3.org/TR/REC-html40/strict.dtd&quot;&gt;
&lt;html&gt;&lt;head&gt;&lt;meta name=&quot;qrichtext&quot; content=&quot;1&quot; /&gt;&lt;style type=&quot;text/css&quot;&gt;
p, li { white-space: pre-wrap; }
&lt;/style&gt;&lt;/head&gt;&lt;body style=&quot; font-family:'Sans Serif'; font-size:9pt; font-weight:400; font-style:normal;&quot;&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;1000&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
        </property>
       </widget>
       <widget class="QLabel" name="label_max_it">
        <property name="geometry">
         <rect>
          <x>20</x>
          <y>80</y>
          <width>41</width>
          <height>21</height>
         </rect>
        </property>
        <property name="text">
         <string>maxit</string>
        </property>
       </widget>
       <zorder>label_L0</zorder>
       <zorder>textEdit_K</zorder>
       <zorder>label_K</zorder>
       <zorder>textEdit_L0</zorder>
       <zorder>label_eps</zorder>
       <zorder>textEdit_eps</zorder>
       <zorder>pbuttonReset</zorder>
       <zorder>textEdit_maxit</zorder>
       <zorder>label_max_it</zorder>
       <zorder>MatplotlibWidget2</zorder>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1255</width>
     <height>20</height>
    </rect>
   </property>
   <property name="styleSheet">
    <string notr="true">QMenuBar{
background: rgb(199, 199, 199)
}</string>
   </property>
   <widget class="QMenu" name="menuExamples">
    <property name="title">
     <string>Examples</string>
    </property>
    <widget class="QMenu" name="menuScotch_collection">
     <property name="title">
      <string>Scotch collection</string>
     </property>
     <addaction name="action3elt_Graph"/>
     <addaction name="action4elt_Graph"/>
     <addaction name="actionCrack"/>
    </widget>
    <addaction name="actionExample_1"/>
    <addaction name="actionExample_2"/>
    <addaction name="separator"/>
    <addaction name="actionExample_3"/>
    <addaction name="separator"/>
    <addaction name="menuScotch_collection"/>
   </widget>
   <addaction name="menuExamples"/>
  </widget>
  <action name="actionExample_1">
   <property name="text">
    <string>Example 1</string>
   </property>
  </action>
  <action name="actionExample_2">
   <property name="text">
    <string>Example 2</string>
   </property>
  </action>
  <action name="actionExample_3">
   <property name="text">
    <string>Example 3</string>
   </property>
  </action>
  <action name="action3elt_Graph">
   <property name="text">
    <string>3elt Graph</string>
   </property>
  </action>
  <action name="action4elt_Graph">
   <property name="text">
    <string>4elt Graph</string>
   </property>
  </action>
  <action name="actionCrack">
   <property name="text">
    <string>Crack</string>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>matplotlibWidget</class>
   <extends>QWidget</extends>
   <header>matplotlibwidged.py</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
